ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -758,19 +766,23 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
                 requestHeader,
                 relativePath,
                 absolutePath,
+                nullptr, /* relativeFromPath */
                 requestHeader->messageType == MessageType_KtoU_NotifyDirectoryPreDelete,  // isDirectory
                 KUMessageTypeToNotificationType(static_cast<MessageType>(requestHeader->messageType)));
             break;
         }
         
         case MessageType_KtoU_NotifyFileHardLinkCreated:
         {
-#if DEBUG
-            // TODO(Mac): Move the following line out of the DEBUG block once we actually need the information. Currently just causes warning-as-error in release build.
-            const char* fromPath = request.paths[MessagePath_From];
-            const char* relativeFromPath = fromPath != nullptr ? GetRelativePath(fromPath, s_virtualizationRootFullPath.c_str()) : nullptr;
-
-            cout << "PrjFSLib.HandleKernelRequest: hard-linked " << (fromPath ?: "[out-of-root path]") << " -> " << (absolutePath ?: "[out-of-root path]") << " (absolute), ";
+            const char* absoluteFromPath = request.paths[MessagePath_From];
+            const char* relativeFromPath = nullptr;
+            if (absoluteFromPath != nullptr)
+            {
+                relativeFromPath = GetRelativePath(absoluteFromPath, s_virtualizationRootFullPath.c_str());
+            }
+            
+#ifdef DEBUG
+            cout << "PrjFSLib.HandleKernelRequest: " << (requestHeader->messageType == MessageType_KtoU_NotifyFileHardLinkCreated ? "hard-linked " : "renamed ");
```

Comment:
There is no need to check `requestHeader->messageType == MessageType_KtoU_NotifyFileHardLinkCreated` because we're in the `MessageType_KtoU_NotifyFileHardLinkCreated` case of the switch statement.

Reply:
Unresolving this conversation, I still see this check:

`requestHeader->messageType == MessageType_KtoU_NotifyFileHardLinkCreated`